app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyController.scala

```diff
@@ -133,22 +120,35 @@ class SubsidyController @Inject() (
       .fallbackTo(Option.empty.toFuture)
   }
 
-  def postReportPayment: Action[AnyContent] = verifiedEmail.async { implicit request =>
-    withLeadUndertaking { undertaking =>
+  def getClaimDate: Action[AnyContent] = verifiedEmail.async { implicit request =>
+    withLeadUndertaking { _ =>
```

Comment:
Note for later - we could move this to an action builder

Reply:
I'm fairly sure there was a reason I didn't use an action builder but I can't really recall why since it was a while ago now. Definitely worth revisiting.